pkg/pool-weighted/test/WeightedPool.test.ts

```diff
@@ -602,9 +609,26 @@ describe('WeightedPool', function () {
     const ZEROS = Array(numberOfTokens).fill(bn(0));
     const weights: BigNumberish[] = WEIGHTS.slice(0, numberOfTokens);
     const initialBalances = INITIAL_BALANCES.slice(0, numberOfTokens);
+    let vault: Contract;
+    let assetManagerContract: Contract;
+    let assetManagers: Array<string>;
 
     async function deployPool(params: RawWeightedPoolDeployment = {}): Promise<void> {
-      params = Object.assign({}, { tokens, weights, swapFeePercentage: POOL_SWAP_FEE_PERCENTAGE, twoTokens }, params);
+      // Deploy Balancer Vault
+      const authorizer = await deploy('v2-vault/Authorizer', { args: [admin.address] });
```

Comment:
I don't understand this entire change - shouldn't we just createa and pass the `assetManagers` array to `WeightedPool.create()`?

Reply:
Do you mean just fill it with ZERO_ADDRESS instead of a real contract? (Was just making the factory and pool tests the same.)

Reply:
Ah I now understand what you did. It is wrong however: `WeightedPool.create()` will deploy a new Vault as well and connect the Pool to that Vault. I'd skip the manual deployment of Authorizer, Vault, AssetManagers, and just pass an EOA to `WeightedPool.create()` as the AM.